exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,3171 @@
+	}
+}
+
+func TestModifyOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.SkipNow()
+	}
+	_, err := ok.ModifyOrder(context.Background(),
+		&order.Modify{
+			AssetType: asset.Spot,
+			Pair:      currency.NewPair(currency.LTC, currency.BTC),
+			OrderID:   "1234",
+			Price:     123456.44,
+			Amount:    123,
+		})
+	if err != nil {
+		t.Errorf("Okx ModifyOrder() error %v", err)
+	}
+}
+
+func TestGetOrderInfo(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("Okx GetOrderInfo() skipping test: api keys not set")
+	}
+	enabled, err := ok.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Error("couldn't find enabled tradable pairs")
+	}
+	if len(enabled) == 0 {
+		t.SkipNow()
+	}
+	_, err = ok.GetOrderInfo(context.Background(),
+		"123", enabled[0], asset.Futures)
+	if err != nil && !strings.Contains(err.Error(), "Order does not exist") {
+		t.Errorf("Okx GetOrderInfo() expecting %s, but found %v", "Order does not exist", err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetDepositAddress(context.Background(), currency.BTC, "", currency.USD.String()); err != nil && !errors.Is(err, errDepositAddressNotFound) {
+		t.Error("Okx GetDepositAddress() error", err)
+	}
+}
+
+func TestWithdraw(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.SkipNow()
+	}
+	withdrawCryptoRequest := withdraw.Request{
+		Exchange: ok.Name,
+		Amount:   0.00000000001,
+		Currency: currency.BTC,
+		Crypto: withdraw.CryptoRequest{
+			Address: core.BitcoinDonationAddress,
+		},
+	}
+	if _, err := ok.WithdrawCryptocurrencyFunds(context.Background(), &withdrawCryptoRequest); err != nil {
+		t.Error("Okx WithdrawCryptoCurrencyFunds() error", err)
+	}
+}
+
+func TestGetPairFromInstrumentID(t *testing.T) {
+	t.Parallel()
+	instruments := []string{
+		"BTC-USDT",
+		"BTC-USDT-SWAP",
+		"BTC-USDT-ER33234",
+	}
+	if _, err := ok.GetPairFromInstrumentID(instruments[0]); err != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", err)
+	}
+	if _, ere := ok.GetPairFromInstrumentID(instruments[1]); ere != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", ere)
+	}
+	if _, erf := ok.GetPairFromInstrumentID(instruments[2]); erf != nil {
+		t.Error("Okx GetPairFromInstrumentID() error", erf)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	pair, err := currency.NewPairFromString("BTC-USD")
+	if err != nil {
+		t.Error(err)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.Limit,
+		Pairs:     currency.Pairs{pair, currency.NewPair(currency.USDT, currency.USD), currency.NewPair(currency.USD, currency.LTC)},
+		AssetType: asset.Spot,
+		Side:      order.Buy,
+	}
+	if _, err := ok.GetActiveOrders(context.Background(), &getOrdersRequest); err != nil {
```

Comment:
I have one order limit, and this is sending 2 requests, maybe we want to check the max return and if it isn't full then we might not need to send a consecutive request.

Reply:
If I get your question right

I think, the parameter *order.GetOrdersRequest doesn't have a field Limit. So, we can't pass the limit; Since the passed value for the method GetOrderList is zero, It will retrieve all the orders and it may not need to call again.

Instead, I added a check to see the number of orders retrieved with the prior call and If the orders are less than 100, I will break out of the loop since the last call retrieved all the remaining.